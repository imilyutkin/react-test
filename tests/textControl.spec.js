import React from 'react';
import {shallow} from 'enzyme';
import TextControl from '../src/components/shared/textControl';
import renderer from 'react-test-renderer';

test('Text control render testing', () =>{
        const textControl = renderer.create(
            <TextControl />
        );

        let tree = textControl.toJSON();
        expect(tree).toMatchSnapshot();
    }
);

test('Text control init value testing', () =>{
        const textControl = renderer.create(
            <TextControl value="hello"/>
        );

        let tree = textControl.toJSON();
        expect(tree).toMatchSnapshot();
    }
);

test('Text control behavior testing', () =>{
        const textControl = renderer.create(
            <TextControl />
        );

        let tree = textControl.toJSON();
        expect(tree).toMatchSnapshot();
    }
);

test('Text control behavior testing', () =>{
        const textControl = shallow(
            <TextControl />
        );
        expect(textControl.find('input').prop('value')).toBe("");
        textControl.find('input').simulate("change", {
            target: {
                value: "hello"
            }
        });
        expect(textControl.find('input').prop('value')).toBe("hello");
    }
);