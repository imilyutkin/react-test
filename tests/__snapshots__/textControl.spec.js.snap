// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text control behavior testing 1`] = `
<input
  onChange={[Function]}
  type="text"
  value=""
/>
`;

exports[`Text control init value testing 1`] = `
<input
  onChange={[Function]}
  type="text"
  value="hello"
/>
`;

exports[`Text control render testing 1`] = `
<input
  onChange={[Function]}
  type="text"
  value=""
/>
`;
